### PR TITLE
Fix issues with markdown parsing and preview being wrong

### LINKED ISSUE #74: [Mentoring/Discussion] Headings of lower levels have styling when writing, but not when saved
I can include from `#` to `####` in the message I am writing. When I write it, it looks styled. When I save, only `#` has any distinct style.

Screen Shot 2021-06-25 at 10 48 26

Screen Shot 2021-06-25 at 10 48 33

### LINKED ISSUE #75: [Mentoring/Discussion] "Unable to save post" when using `#####`
When I attempt to add `#####` in my comment, I get this generic unstyled error. Note that the error calls the record a "post" but the user is writing a "comment".

Screen Shot 2021-06-25 at 10 50 44

### PR DESCRIPTION
**Pull request by iHiD**
Fixes https://github.com/exercism/v3-beta/issues/75
Fixes https://github.com/exercism/v3-beta/issues/74

### CODE REVIEW

File: test/commands/markdown/parse_test.rb [hunk 1]
```diff
@@ -162,6 +162,11 @@ class Markdown::ParseTest < ActiveSupport::TestCase
     assert_equal "<h3>Level two</h3>\n<h4>Level three</h4>\n", Markdown::Parse.("## Level two\n\n### Level three")
   end
 
+  test "does not lower headings beyond h6" do
+    str = "#### Level four\n\n##### Level five\n\n###### Level six\n\n####### Level seven"
+    assert_equal "<h5>Level four</h5>\n<h6>Level five</h6>\n<h6>Level six</h6>\n<p>####### Level seven</p>\n", Markdown::Parse.(str)
+  end
```
Comment (iHiD) [inline test/commands/markdown/parse_test.rb:168]: @ErikSchierboom Could you check this and the corrosponding code pls?

File: app/controllers/api/markdown_controller.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 module API
   class MarkdownController < BaseController
     def parse
-      html = Markdown::Parse.(params[:markdown].to_s)
-
+      opts = params.permit!.to_hash.with_indifferent_access[:parse_options]
```
Comment (ErikSchierboom) [inline app/controllers/api/markdown_controller.rb:4]: What does the `permit!` call do?

**Review by ErikSchierboom — APPROVED** [review]
LGTM

File: app/controllers/api/markdown_controller.rb (hunk 2 above)
Comment (iHiD) [reply app/controllers/api/markdown_controller.rb:4]: Tells StrongParamaters to fuck off :)

**Pull request merged**